File: returned/python/today3/group_anagrams.py
# https://leetcode.com/problems/group-anagrams/description/
from typing import List

class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        result_dict = {}
        for s in strs:
            char_list = list(s)
            char_list.sort()
            sorted_str = ''.join(char_list)
            if sorted_str in result_dict:
                result_dict[sorted_str].append(s)
            else:
                result_dict[sorted_str] = [s]       
        return list(result_dict.values())

File: typescript/solved/group-anagrams.ts
/**
 * https://leetcode.com/problems/group-anagrams/
 */
export function groupAnagrams(strs: string[]): string[][] {
  const grouped: Record<string, string[]> = {};
  for (let s of strs) {
    const sorted = s.split('').sort().join();
    if (!grouped[sorted]) grouped[sorted] = [];
    grouped[sorted].push(s);
  }
  return Object.values(grouped);
}
